### ISSUE TITLE
Running `pod spec lint` causes an "unexpected end of file" error in generated scripts

### ISSUE THREAD

**Issue by reidmain** [CocoaPods/CocoaPods#5331]
🌈

# Report

## What did you do?

I was attempting to update my pod [FDKeychain](https://github.com/reidmain/FDKeychain) to support watchOS. I wanted to validate the Podspec before I pushed all my changes. I ran `pod spec lint FDKeychain.podspec`.

## What did you expected to happen?

The Podpsec would validate successfully because all I did was add a new deployment target.

## What happened instead?

The validation failed with some strange errors inside some scripts that CocoaPods generated.

```
 -> FDKeychain (1.2.2)
    - ERROR | xcodebuild: Returned an unsuccessful exit code. You can use `--verbose` for more information.
    - NOTE  | [iOS] xcodebuild:  /Users/Reid/Library/Developer/Xcode/DerivedData/App-ewluisdblbkzrffvqaxxpmxteysf/Build/Intermediates/App.build/Release-iphonesimulator/App.build/Script-3AC2808C4D5BAEEB37313CD4.sh: line 10: syntax error: unexpected end of file
    - NOTE  | [OSX] xcodebuild:  /Users/Reid/Library/Developer/Xcode/DerivedData/App-ewluisdblbkzrffvqaxxpmxteysf/Build/Intermediates/App.build/Release/App.build/Script-8F7D060E8E030F32C4E34C2E.sh: line 10: syntax error: unexpected end of file
    - NOTE  | [watchOS] xcodebuild:  /Users/REid/Library/Developer/Xcode/DerivedData/App-ewluisdblbkzrffvqaxxpmxteysf/Build/Intermediates/App.build/Release-watchsimulator/App.build/Script-7A2C21E68397F4054B0DFE0A.sh: line 10: syntax error: unexpected end of file

Analyzed 1 podspec.
```

Below is the output of the iOS script:
```
#!/bin/sh
diff "${PODS_ROOT}/../Podfile.lock" "${PODS_ROOT}/Manifest.lock" > /dev/null
						if [[ $? != 0 ]] ; then
						    cat << EOM
						error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.
						EOM
						    exit 1
						fi
						
```
### Stack

```
   CocoaPods : 1.0.0
        Ruby : ruby 2.0.0p353 (2013-11-22 revision 43784) [x86_64-darwin13.3.0]
    RubyGems : 2.6.4
        Host : Mac OS X 10.11.4 (15E65)
       Xcode : 7.3.1 (7D1014)
         Git : git version 2.8.2
Ruby lib dir : /Users/Reid/.rbenv/versions/2.0.0-p353/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ 11e493632bd6a8d2e23fd60882a16335839730a8
```

### Installation Source

```
Executable Path: /Users/Reid/.rbenv/versions/2.0.0-p353/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.0.0
cocoapods-try         : 1.0.0
```

## Project that demonstrates the issue

I'd say the first step would be to clone https://github.com/reidmain/FDKeychain and try running `pod spec lint FDKeychain.podspec` and see if you get a similar error. I assume that it has something to do with my installation of CocoaPods and isn't a Podspec issue because I used `pod spec create` and modified it so it was similar to my current Podspec and it also failed.

**Comment by reidmain** [comment]
I attempted to downgrade to CocoaPods 0.39.0 and the error I now see is:
```
 -> FDKeychain (1.2.2)
    - ERROR | [iOS] unknown: Encountered an unknown error (Xcodeproj relies on a library called `fiddle` to read and write Xcode project files. Ensure your Ruby installation includes `fiddle` and try again.) during validation.
```

I saw this error when I first attempted to validate the Podspec using CocoaPods 1.0.0 however I solved it by running `gem install fiddle`. That didn't seem to solve the problem this time.

Also `pod env` seems to be a 1.0.0 feature so I can't give the output of that again. But I am still using everything that was listed above. I didn't use rbenv to change anything or modify Xcode in any way.

**Comment by neonichu** [comment]
Looks like the generated script is somehow getting indented weirdly, also fiddle should be part of your Ruby installation. I would recommend reinstalling Ruby or using the CocoaPods app, since there seems to be something wrong with your current installation.

**Comment by reidmain** [comment]
The error message did not make it clear that "fiddle" was a part of Ruby and not the "fiddle" gem https://rubygems.org/gems/fiddle. Perhaps the error messaging should be updated to indicate that.

I uninstalled all of my gems as well as version 2.0.0-p353 of Ruby that rbenv was using. I then installed 2.0.0-p648 of Ruby using rbenv and reinstalled CocoaPods v1.0.0 and `pod spec lint` passed. Perhaps there is something off with fiddle in 2.0.0-p353 of Ruby.

Thank you for your help @neonichu
